cilib.sh

```diff
@@ -102,3 +102,13 @@ print(unit[units[0]]['public-address'])
 scriptPath() {
     env python3 -c "import os,sys; print(os.path.dirname(os.path.abspath(\"$0\")))"
 }
+
+ci_lxc_launch()
+{
+    # Launch local LXD container to publish to charmcraft
+    local lxc_image=$1
+    local lxc_container=$2
+    sudo lxc launch ${lxc_image} ${lxc_container}
+    sleep 10
```

Comment:
Perhaps this should use the `until` loop pattern instead?

Reply:
There was concern that the `sleep 10` was giving the container time to handle the incoming shell commands afterwards.  Without it -- `lxc shell` would hang perpetually.  The `until` would never fail and never cause a retry b/c it was blocked waiting on a result from `lxc shell` which wouldn't come back.

are you thinking

```bash
sleep 10
sudo lxc shell ${lxc_container} -- bash -c "apt-get update && apt-get install build-essential -y"
```

becomes


```bash
sleep 10
until sudo lxc shell ${lxc_container} -- bash -c 'apt-get update && apt-get install build-essential -y'; do
  sleep 3
  echo 'retrying install of build-essentials'
done
```

Reply:
Ah, right.  I was only half paying attention to that discussion.  I'm just not a fan of arbitrary sleeps, in general. If it works, that's fine.  If it becomes a problem in the future (such as on a slow runner), we can explore things like a `timeout` wrapper on the `lxc shell` or something.